interpreter/binary/encode.ml

```diff
@@ -477,39 +477,48 @@ let encode m =
       section 10 (vec code) fs (fs <> [])
 
     (* Element section *)
-    let segment active passive seg =
-      match seg.it with
-      | Active {index; offset; init} ->
-        if index.it = 0l then
-          u8 0x00
-        else begin
-          u8 0x02; var index
-        end;
-        const offset; active init
-      | Passive {etype; data} ->
-        u8 0x01; passive etype data
-
-    let active_elem el =
+    let elem_expr el =
+      match el.it with
+      | RefNull -> u8 0xd0; end_ ()
+      | RefFunc x -> u8 0xd2; var x; end_ ()
+
+    let elem_index el =
       match el.it with
       | RefNull -> assert false
       | RefFunc x -> var x
 
-    let passive_elem el =
-      match el.it with
-      | RefNull -> u8 0xd0; end_ ()
-      | RefFunc x -> u8 0xd2; var x; end_ ()
+    let elem_indices data = vec elem_index data
 
     let table_segment seg =
-      let active init = vec active_elem init in
-      let passive etype data = elem_type etype; vec passive_elem data in
-      segment active passive seg
+      match seg.it with
+      | ElemActive {index = {it = 0l;_}; offset; init; _}
+        when not (contains_null_ref init) ->
```

Comment:
We will add more expressions in the future, so a more forward-compatible predicate would be `all_func_ref`.

Reply:
Done.